src/main/kotlin/uk/gov/gdx/datashare/services/EventDataService.kt

```diff
@@ -123,6 +127,15 @@ class EventDataService(
     return mapEventNotification(event, acquirerSubscription, emptyList(), false)
   }
 
+  @Scheduled(fixedRate = 1, timeUnit = TimeUnit.HOURS)
+  @SchedulerLock(name = "countUnconsumedEvents", lockAtMostFor = "3m", lockAtLeastFor = "3m")
+  fun countUnconsumedEvents() {
+    LockAssert.assertLocked()
+    log.debug("Looking for unconsumed events")
+    val unconsumedEventsCount = eventDataRepository.findAll().count()
```

Comment:
Should be able to just get a count. This will return all the data, create a JVM object for each, then count the objects. We can just ask postgres for a number 🙂